includes/class-wc-gateway-stripe.php

```diff
@@ -650,6 +611,10 @@ public function process_payment( $order_id, $retry = true, $force_save_source =
 			WC_Stripe_Logger::log( "Info: Begin processing payment for order $order_id for the amount of {$order->get_total()}" );
 
 			$intent = $this->get_intent_from_order( $order );
+			if ( 'setup_intent' === $intent->object ) {
+				$intent = false; // This function can only deal with *payment* intents
```

Comment:
🤔 The only case where an order has a SetupIntent *and* `process_payment` will be called is a Pre-Order. In that case, `process_payment` calls `process_pre_order()` and returns before this line. So no, this is not needed. However, since `get_intent_from_order()` *can* return a SetupIntent and this function assumes it's a PaymentIntent, I think it's a bit clearer to make that condition explicit.

What do you think?

Reply:
I agree with you, this is better and more readable. Let's leave it.

Especially considering the explicitness I requested in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1024#discussion_r330029773 :D